nautobot/utilities/git.py

```diff
@@ -8,17 +8,43 @@
 
 logger = logging.getLogger("nautobot.utilities.git")
 
+# 'A' and 'D' status are swapped because of the way the repo.git.diff was implemented
+# e.g. 'A' actually stands for Addition but in this case is Deletion
+GIT_STATUS_MAP = {
+    "A": "Deletion",
+    "M": "Modification",
+    "C": "Copy",
+    "D": "Addition",
+    "R": "Renaming",
+    "T": "File Type Changed",
+    "U": "File Unmerged",
+    "X": "Unknown",
+}
+
+
+def swap_status_initials(x):
+    """Swap github status initials with its equivalent"""
+    return GIT_STATUS_MAP.get(x[0]) + " - " + x[1]
+
 
 class BranchDoesNotExist(Exception):
     pass
 
 
 class GitRepo:
-    def __init__(self, path, url):
+    def __init__(self, path, url, init_repo=False):
         """
         Ensure that we have a clone of the given remote Git repository URL at the given local directory path.
+
+        Args:
+            path (str): path to git repo
+            url (str): git repo url
+            init_repo (bool): determines if a local git repository should be initialized or not
         """
-        if os.path.isdir(path):
+        if init_repo:
+            self.repo = Repo.init(path)
+            self.repo.create_remote("origin", url=url)
+        elif os.path.isdir(path):
             self.repo = Repo(path=path)
         else:
             self.repo = Repo.clone_from(url, to_path=path)
```

Comment:
I'm still not quite happy with the logic here, it feels unintuitive, and I don't like that `init_repo` overrides the handling of an existing Git repository at `path`. Maybe instead we could invert the flag and revise the logical branches, something like:

```python
def __init__(self, path, url, clone_initially=True):
    """..."""
    if os.path.isdir(path):
        self.repo = Repo(path=path)
    elif clone_initially:
        self.repo = Repo.clone_from(url, to_path=path)
    else:
        self.repo = Repo.init(path)
        self.repo.create_remote("origin", url=url)
```

Reply:
Thanks